### ISSUE TITLE
Insufficient contrast between links and surrounding text

### ISSUE THREAD

**Issue by cimaja** [microsoft/fluentui#16302]
### Environment Information

- **Package version(s)**: 7.155.3
- **Browser and OS versions**: Edge(chromium) 85.0.564.63, Windows 10 Enterprise 10.0.19042 Build 19042

### Describe the issue:

Links should have contrast ratio of 3:1 with surrounding text and have a non-color visual distinction for colorblind people ([see W3C for more details](https://www.w3.org/WAI/WCAG21/Techniques/failures/F73.html)).

Find below some Link/Text contrast ratios for various themes:
- [With Default theme, contrast is 2.86:1](https://webaim.org/resources/contrastchecker/?fcolor=0078D4&bcolor=323130)
- [With Power Apps theme, contrast is 1.41:1](https://webaim.org/resources/contrastchecker/?fcolor=742774&bcolor=323130)
- [With Power Automate theme, contrast is 2.68:1](https://webaim.org/resources/contrastchecker/?fcolor=0066FF&bcolor=323130)

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/cimaja/pen/yLapMXB alternatively the issue is also visible on [the Fluent demo website. ](https://fabricweb.z5.web.core.windows.net/oufr/7.155.3/#/examples/link)

#### Actual behavior:

Fluent links use only color to differentiate links from surrounding text. 

#### Expected behavior:

Fluent links should be underlined to help differentiate links from normal text for color-blind people. Optionally, links color should have a contrast ratio greater than 3:1 with surrounding text.

### Documentation describing expected behavior
[Failure of Success Criterion 1.4.1 due to creating links that are not visually evident without color vision](https://www.w3.org/WAI/WCAG21/Techniques/failures/F73.html)

**Comment by ecraig12345** [comment]
Talked about this with our designers, and the proposed solution (which aligns with an upcoming design refresh) is to add a new prop determining whether to use an underline or not. The terms used for this in the design refresh will be "subtle" (no underline) and "inline" (underline).

This is my proposal for the API but open to suggestions from @khmakoto (who's worked on Link convergence) or others.
```ts
/**
 * Determines which styling variant to use.
 * `inline` means the link will be underlined, suitable for use within text.
 * `subtle` means the link is not underlined, suitable for standalone use.
 * 
 * The default is currently `subtle` to match existing behavior, but will likely change to 
 * `inline` in a future release.
 * @default 'subtle'
 */
variant?: 'inline' | 'subtle';
```

**Comment by ecraig12345** [comment]
Or I suppose a simpler solution would be to just do a boolean prop `underlined`.

**Comment by khmakoto** [comment]
@ecraig12345, that's what I was about to say, have a boolean `underlined` or `inline` prop with default value `false` so we can default to no underline.

**Comment by ecraig12345** [comment]
@khmakoto Hmm I could go with either of those. `inline` as a standalone prop name is a little overloaded since there's also CSS `display: inline`, though actually that might actually work to our advantage in this case since a `display: inline` link ought to also be underlined. `underlined` is longer but entirely unambiguous.

**Comment by khmakoto** [comment]
After speaking with @ecraig12345 offline we'll go with `underline` for the moment and possibly revisit in the future. I'll work on the change as soon as I can.

**Comment by ecraig12345** [comment]
@khmakoto Just realized, we should ideally also update the examples which use Links within text to use the new prop (definitely the Link examples, maybe others too if it's not too hard). And update `MarkdownLink` in `react-docsite-components`/`example-app-base` to be underlined by default.
